### PR TITLE
Documentation for Prow job configuration and processes

### PR DESCRIPTION
**Pull request by RedbackThomson**
Add a document explaining how the Prow jobs are configured and the structure of our repository. 
Another document outlining the IAM process is still in draft.

### CODE REVIEW

File: docs/prow-jobs.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+## Postsubmit Jobs
+
+Currently, the only postsubmit job we manage with Prow is to publish any new
+versions of the service controllers into the [ECR public repository][ecr-repo]
+and into the [Helm chart repository][helm-repo].
+
+The Dockerfile we use for continuous deployment is
+`prow/jobs/images/Dockerfile.deploy`. It is based off `buildah`, which we use to
+construct OCI-compliant images.
```
Comment (jaypipes) [inline docs/prow-jobs.md:68]: Well, specifically, we use `buildah` in order to produce OCI images *from within another container*. :)

File: docs/prow-jobs.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+[testing-docs]: https://aws-controllers-k8s.github.io/community/dev-docs/testing/
+
+The Dockerfile we use for integration tests is
+`prow/jobs/images/Dockerfile.test`. It includes all the command line tools 
+required to run our bootstrapping and testing scripts. The image also includes a
+full version of the Docker engine, which serves as the container runtime for
+KIND and our Python e2e test container. The entrypoint for the image is a shell
+script that optionally enables the Docker engine and assumes the appropriate
+test role within the pod. For more information on the IAM pathway, see the
+[iam-structure](iam-structure.md) document.
```
Comment (jaypipes) [inline docs/prow-jobs.md:52]: did you forget to `git add` this file?

**Review by jaypipes — COMMENTED** [review]
Excellent work on this @RedbackThomson. Couple things inline but overall awesome.

File: docs/prow-jobs.md (hunk 2 above)
Comment (RedbackThomson) [reply docs/prow-jobs.md:52]: Nope. The IAM story is still a work in progress. We don't want every test's resources to be in the Prow account, so am still working on assuming a role in the service team's accounts to do so.

**Pull request merged**